source/arch/riscv/register-node.cpp

```diff
@@ -13,19 +13,23 @@
  * GNU General Public License for more details.
  *
  * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.*/
+* along with this program. If not, see <http://www.gnu.org/licenses/>.*/
 
-#ifndef ERAGPSIM_ARCH_COMMON_INSTRUCTION_ASSEMBLER_HPP
-#define ERAGPSIM_ARCH_COMMON_INSTRUCTION_ASSEMBLER_HPP
+#include "arch/riscv/register-node.hpp"
+#include "arch/riscv/utility.hpp"
+#include "common/assert.hpp"
 
-#include <functional>
-#include <vector>
+namespace riscv {
 
-#include "arch/common/instruction-key.hpp"
-#include "arch/riscv/instruction-node.hpp"
-#include "core/memory-value.hpp"
+RegisterNode::RegisterNode(const std::string& identifier)
+: AbstractRegisterNode(identifier) {
+}
 
-using AssemblerFunction =
-    std::function<std::vector<bool>(InstructionKey, std::vector<MemoryValue>)>;
+MemoryValue RegisterNode::assemble() const {
+  // all register identifiers of riscv start with x
```

Comment:
We should probably store the `RegisterInformation` and not the identifier only, to have the numeric ID. Or just the numeric ID